### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by StefMa**
* Move trello API to its own package
* Use html/template for rendering

> **Do not merge** until the integration hackathon is over..

### CODE REVIEW

File: api/trello/api.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  "net/http"
+  "encoding/json"
+  "io/ioutil"
+)
+
+const API_KEY = "[API_KEY]"
+
+var LinkForAuthKey = "https://trello.com/1/authorize?expiration=never&scope=read,write&response_type=token&name=Treit&key=" + API_KEY
+
+func CreateNewBoard(authKey string, boardname string) (board Board) {
```
Comment (Seept) [inline api/trello/api.go]: Also return the error here. The user of the package should handle the error.

Comment (Seept) [reply api/trello/api.go]: I would return a pointer to Board, so that you can return nil if an error occursed.

File: api/trello/api.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  response, err := http.Post("https://api.trello.com/1/boards?name=" + boardname + "&key=" + API_KEY + "&token=" + authKey, "", nil)
+  if err != nil {
+    // TODO: Handle error
```
Comment (Seept) [inline api/trello/api.go:29]: return nil, err

File: api/trello/api.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  }
+  defer response.Body.Close()
+  body, err := ioutil.ReadAll(response.Body)
+  if err != nil {
+    // TODO: Handle error
+  }
+  board = Board{}
+  json.Unmarshal(body, &board)
```
Comment (Seept) [inline api/trello/api.go:37]: Unmarshal also returns an error

File: api/trello/api.go [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  return
```
Comment (Seept) [inline api/trello/api.go:38]: Does this actually work ? This should not compile

File: api/trello/api.go [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+  body, err := ioutil.ReadAll(response.Body)
+  if err != nil {
+    // TODO: Handle error
+  }
+  json.Unmarshal(body, &lists)
+  return
+}
+
+func MoveCardToList(authKey string, cardId string, listId string) {
+  client := &http.Client{}
```
Comment (Seept) [inline api/trello/api.go]: I would Create a TrelloClient struct which contains the authKey as a field. `Reuse` the client object.

func (client *TrelloClient) MoveCardtoList(cardId string listId, string)

Comment (Seept) [reply api/trello/api.go]: Ofc for all your methods here.

File: api/trello/api.go (hunk 4 above)
Comment (StefMa) [reply api/trello/api.go:38]: Why shouldn't it work? 🤔 
This feature is [named return values](https://tour.golang.org/basics/7)..

Comment (Seept) [reply api/trello/api.go:38]: ah.. got it.. You are right. I just never used this feature.

**Pull request merged**